src/main/java/reposense/commits/CommitResultAggregator.java

```diff
@@ -98,19 +103,19 @@ private static float getContributionVariance(List<AuthorDailyContribution> contr
     }
 
     private static Map<Author, List<AuthorDailyContribution>> getAuthorDailyContributionsMap(
-            Set<Author> authorSet, List<CommitResult> commitResults, String zoneId) {
+            Set<Author> authorSet, List<CommitResult> commitResults, ZoneId zoneId) {
         Map<Author, List<AuthorDailyContribution>> authorDailyContributionsMap = new HashMap<>();
         authorSet.forEach(author -> authorDailyContributionsMap.put(author, new ArrayList<>()));
 
-        Date commitStartDate = null;
+        LocalDateTime commitStartDate;
         for (CommitResult commitResult : commitResults) {
-            commitStartDate = getSystemStartOfDate(commitResult.getTime(), zoneId);
+            commitStartDate = getStartOfDate(commitResult.getTime(), zoneId);
             Author commitAuthor = commitResult.getAuthor();
 
             List<AuthorDailyContribution> authorDailyContributions = authorDailyContributionsMap.get(commitAuthor);
 
-            if (authorDailyContributions.isEmpty()
-                    || !authorDailyContributions.get(authorDailyContributions.size() - 1).getDate()
+            if (authorDailyContributions.isEmpty() || !getStartOfDate(authorDailyContributions
+                    .get(authorDailyContributions.size() - 1).getDate(), zoneId)
                             .equals(commitStartDate)) {
```

Comment:
Let's create a new boolean variable that describes the second part of this condition, because I think this is too long and not easily readable.

Reply:
This is what I currently have for the refactoring:
![Screenshot (82)](https://user-images.githubusercontent.com/69678785/152947105-50674249-d42d-4009-afee-b3dc9bd7da3a.png)

Personally, I think that splitting the `addDailyContributionForNewDate` into nested if-else blocks isn't ideal. I can't put the boolean in front since there would be an invalid `authorDailyContributions.get(-1)` for empty `authorDailyContributions` before the `isEmpty()` check is done.

What I suggest instead is to put comments for the boolean check, like:
```
// Checks whether there are no contribution dates present 
// or the most recently inserted contribution date does not match commit date
```